src/explanation.rs

```diff
@@ -0,0 +1,184 @@
+    /// of the prompt should be indicated)
+    pub target: &'a str,
+    /// Granularity paramaters for the explanation
+    pub granularity: Granularity,
+}
+
+/// Granularity parameters for the [TaskExplanation]
+#[derive(Default)]
+pub struct Granularity {
+    /// The granularity of the parts of the prompt for which a single
+    /// score is computed.
+    prompt: PromptGranularity,
+}
+
+impl Granularity {
+    /// Returns a new [Granularity] based on the given one with the [Granularity::prompt]
+    /// being set to `prompt_granularity`.
+    pub fn with_prompt_granularity(self, prompt_granularity: PromptGranularity) -> Self {
+        Self {
+            prompt: prompt_granularity,
+        }
+    }
+}
+
+/// At which granularity should the target be explained in terms of the prompt.
+/// If you choose, for example, [PromptGranularity::Sentence] then we report the importance score of each
+/// sentence in the prompt towards generating the target output.
+/// The default is [PromptGranularity::Auto] which means we will try to find the granularity that
+/// brings you closest to around 30 explanations. For large documents, this would likely
+/// be sentences. For short prompts this might be individual words or even tokens.
+#[derive(Serialize, Copy, Clone, PartialEq, Default)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum PromptGranularity {
+    /// Let the system decide which granularity is most suitable for the given input.
+    /// This will result
+    #[default]
+    Auto,
+    Word,
+    Sentence,
+    Paragraph,
+}
+
+impl PromptGranularity {
+    fn is_auto(&self) -> bool {
+        self == &PromptGranularity::Auto
+    }
+}
+
+/// Body sent to the Aleph Alpha API for an explanation request
+#[derive(Serialize)]
+struct BodyExplanation<'a> {
+    prompt: Prompt<'a>,
+    target: &'a str,
+    #[serde(skip_serializing_if = "PromptGranularity::is_auto")]
+    prompt_granularity: PromptGranularity,
+    model: &'a str,
+}
+
+/// Body received by the Aleph Alpha API from an explanation request
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct ResponseExplanation {
+    /// The Body contains an array of [Explanation]s one for each
+    /// part of the target being explained.
+    explanations: Vec<Explanation>,
+}
+
+/// The result of an explanation request.
+#[derive(Debug, PartialEq)]
+pub struct ExplanationOutput {
+    /// Since at the moment we only
+    /// support explaining the target as a whole the output only contains
+    /// a single [Explanation].
+    pub explanation: Explanation,
+}
+
+impl ExplanationOutput {
+    fn from(mut response: ResponseExplanation) -> ExplanationOutput {
+        ExplanationOutput {
+            explanation: response.explanations.pop().unwrap(),
+        }
+    }
+}
+
+/// The explanation for a specific part of the target.
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct Explanation {
+    /// The part of the target that is explained
+    pub target: String,
```

Comment:
Regarding the comment on the output, if we are really only going to handle the case of 1 completion, I think we should either go all in, or support target granularity.

If we only support the complete target granularity (which I think is a fair choice) then the output could be 

```rust
pub struct ExplanationOutput {
    pub items: Vec<ItemExplanation>,
}
```

The `target` field is useless unless you want to distinguish between multiple parts of a target. So we can just give the user the scores without the extra struct boilerplate